compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -1271,7 +1287,16 @@ pub(super) fn hir_module_items(tcx: TyCtxt<'_>, module_id: LocalDefId) -> Module
         }
 
         fn visit_item(&mut self, item: &'hir Item<'hir>) {
+            if associated_body(Node::Item(item)).is_some() {
+                self.body_owners.push(item.def_id);
+            }
+
             self.items.push(item.item_id());
+
+            if self.tcx.hir().is_body_owner(item.def_id) {
+                self.body_owners.push(item.def_id);
```

Comment:
This probably pushes the same `LocalDefId` twice.

Reply:
Thanks, that was the problem.